Python/compile.c

```diff
@@ -4918,25 +4948,44 @@ stackdepth_walk(struct compiler *c, basicblock *b, int depth, int maxdepth)
             maxdepth = depth;
         assert(depth >= 0); /* invalid code or bug in stackdepth() */
         if (instr->i_jrel || instr->i_jabs) {
+            /* Recursively inspect jump target */
             target_depth = depth;
+            if (instr->i_opcode == CONTINUE_LOOP) {
+                /* Pops variable number of values from the stack,
+                 * but the target should be already proceeding.
+                 */
+                assert(instr->i_target->b_seen);
+                assert(instr->i_target->b_startdepth <= depth);
+                goto out; /* remaining code is dead */
+            }
             if (instr->i_opcode == FOR_ITER) {
-                target_depth = depth-2;
+                target_depth = depth - 2;
             }
             else if (instr->i_opcode == SETUP_FINALLY ||
-                     instr->i_opcode == SETUP_EXCEPT) {
-                target_depth = depth+3;
-                if (target_depth > maxdepth)
-                    maxdepth = target_depth;
+                     instr->i_opcode == SETUP_EXCEPT)
+            {
+                depth = depth - 6;
```

Comment:
It's not obvious why that is. `SETUP_FINALLY` and `SETUP_EXCEPT` merely call `PyFrame_BlockSetup`. Can you add a comment?

Reply:
PyFrame_BlockSetup doesn't affect the stack of values. The stack effect of these opcodes is 0 in normal flow.

Reply:
Can you add comments for all this? (why -6?)

Reply:
`+6 (jump) - 6 = 0 (not jump)`

Do you want to duplicate comments from PyCompile_OpcodeStackEffect()? Or just add a reference to PyCompile_OpcodeStackEffect()?

Reply:
Duplicating comments is fine. The file is large anyway.

Reply:
In any case I'm going to rewrite this code in bpo-32455 to something like:
```c
new_depth = depth + PyCompile_OpcodeStackEffectEx(opcode, oparg, 0);
if (new_depth > maxdepth)
    maxdepth = new_depth;
if (isjump) {
    target_depth = depth + PyCompile_OpcodeStackEffectEx(opcode, oparg, 1);
    maxdepth = stackdepth_walk(c, instr->i_target, target_depth, maxdepth);
}
depth = new_depth;
```
And all comments will be in one place.